### PR TITLE
Refactor scripts and update locust to v1

### PR DESCRIPTION
**Pull request by JamesGardiner**
### What is the context of this PR?
Updates locust

### How to review 
Create a new virtual environment with `pipenv install --dev`. Run the benchmark against a local runner instance using `make run`. Check the output is as expected.

### CODE REVIEW

**Comment by JamesGardiner** [comment]
[Profile Runner and identify/document expensive calls times (s)](https://trello.com/c/ReffUmOH/4007-profile-runner-and-identify-document-expensive-calls-times-s)

File: run.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 echo "Running Locust benchmark"
-USER_WAIT_TIME_MIN_SECONDS=0 USER_WAIT_TIME_MAX_SECONDS=0 REQUESTS_JSON="${1}" HOST="${2:-http://localhost:5000}" pipenv run locust --no-web -c 1 -r 1 -t ${3:-1m} --csv=${4:-output} -L WARNING
+USER_WAIT_TIME_MIN_SECONDS=0 USER_WAIT_TIME_MAX_SECONDS=0 REQUESTS_JSON="${1}" HOST="${2:-http://localhost:5000}" pipenv run locust --headless -u 1 -r 1 -t ${3:-1m} --csv=${4:-output} -L WARNING
```
Comment (ajmaddaford) [inline run.sh:4]: These changes will need to be tested in GCP as the Helm scripts will need updating - https://github.com/ONSdigital/eq-survey-runner-benchmark/blob/master/k8s/helm/values.yaml#L14

File: locustfile.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-from locust import HttpLocust
+from locust import constant, HttpUser
 
 from runner_benchmark.taskset import SurveyRunnerTaskSet
 
 
-class SurveyRunnerScenario(HttpLocust):
-    task_set = SurveyRunnerTaskSet
+class SurveyRunnerScenario(HttpUser):
+    wait_time = constant(0)
```
Comment (ajmaddaford) [inline locustfile.py:9]: We should review our current use of sleep for a wait time. Happy for this to be done in a separate card.

**Comment by ajmaddaford** [comment]
I ran this via a pipeline and while the test ran successfully, the posting to Slack failed. Running the `visualise_results` scripts locally generates the same error:

```
OUTPUT_DIR="outputs/daily-test" \
NUMBER_OF_DAYS="7" \
pipenv run python -m scripts.visualise_results
Traceback (most recent call last):
  File "/Users/andrew/.pyenv/versions/3.7.6/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/andrew/.pyenv/versions/3.7.6/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/andrew/dev/eq-survey-runner-benchmark/scripts/visualise_results.py", line 24, in <module>
    results = get_results(folders, parsed_variables['number_of_days'])
  File "/Users/andrew/dev/eq-survey-runner-benchmark/scripts/get_summary.py", line 37, in get_results
    percentile_99th = int(values[18])
IndexError: list index out of range
```

**Comment by JamesGardiner** [comment]
@ajmaddaford I've changed the script to use CSV reader, which should guard against future changes such as the one that has caused this issue (Locust now adds an empty new line to it's csv output).

File: scripts/benchmark_stats.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class BenchmarkStats:
+    def __init__(self, folder_path: str):
+        self._files: List[str] = glob(folder_path + "/*stats.csv")
+        self.get_requests: List[int] = []
+        self.post_requests: List[int] = []
+        self.total_requests: int = 0
+        self.total_failures: int = 0
+        self.error_percentage: int
```
Comment (MebinAbraham) [inline scripts/benchmark_stats.py:14]: Isn't needed given it is never set.

File: scripts/benchmark_stats.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    if "/questionnaire" in row['Name']:
+                        if row["Type"] == "GET":
+                            self.get_requests.append(percentile_99th)
+                        elif row["Type"] == "POST":
+                            self.post_requests.append(percentile_99th)
+
+                    if row["Name"] == "Aggregated":
+                        self.total_requests += int(row["Request Count"])
+                        self.total_failures += int(row["Failure Count"])
+
```
Comment (MebinAbraham) [inline scripts/benchmark_stats.py:41]: Need to handle old format, as existing files cannot be done using this. i.e Daily test stats will fail. We should look at archiving the current benchmark data in GCS if we didn't wsnt to handle the old data.

File: scripts/benchmark_stats.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                    percentile_99th = int(row["99%"])
+                    if "/questionnaire" in row['Name']:
+                        if row["Type"] == "GET":
+                            self.get_requests.append(percentile_99th)
+                        elif row["Type"] == "POST":
+                            self.post_requests.append(percentile_99th)
+
+                    if row["Name"] == "Aggregated":
+                        self.total_requests += int(row["Request Count"])
+                        self.total_failures += int(row["Failure Count"])
```
Comment (MebinAbraham) [inline scripts/benchmark_stats.py:40]: Update to my comment about backwards compatibility. I think they should be backwards compatible given the only change needed are these:
```suggestion
                        request_count = row.get("Request Count", row.get("# requests"))
                        failure_count = row.get("Failure Count", row.get("# failures"))
                        self.total_requests += int(request_count)
                        self.total_failures += int(failure_count)
```

File: scripts/benchmark_stats.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+from csv import DictReader
+from glob import glob
+from statistics import mean
+from typing import Generator, Iterable, List
+
+
+class BenchmarkStats:
+    def __init__(self, folder_path: str):
+        self._files: List[str] = glob(folder_path + "/*stats.csv")
```
Comment (MebinAbraham) [inline scripts/benchmark_stats.py:9]:
```suggestion
        self._files: List[str] = glob(f"{folder_path}/*stats.csv")
```

File: scripts/benchmark_stats.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+                    if row["Name"] == "Aggregated":
+                        request_count = row.get("Request Count", row["# requests"])
+                        failure_count = row.get("Failure Count", row["# failures"])
+                        self.total_requests += int(request_count)
+                        self.total_failures += int(failure_count)
+
+    @property
+    def files(self) -> Generator[str, None, None]:
+        return self._files
```
Comment (MebinAbraham) [inline scripts/benchmark_stats.py:45]: The type hinting mismatches. One is a list, one is a generator. `self._files: List[str] = glob(folder_path + "/*stats.csv")`

Comment (JamesGardiner) [reply scripts/benchmark_stats.py:45]: Good catch. Did your IDE get this or did you?

Comment (MebinAbraham) [reply scripts/benchmark_stats.py:45]: IDE

File: scripts/get_summary.py [hunk 8]
```diff
@@ -1,33 +1,32 @@
...
+from typing import List, NamedTuple
 
+from scripts.benchmark_stats import BenchmarkStats
 
-def get_results(folders, number_of_days=None):
-    results_list = []
 
+class Result(NamedTuple):
+    date: str
+    statistics: List[str]
```
Comment (MebinAbraham) [inline scripts/get_summary.py:13]:
```suggestion
    statistics: BenchmarkStats
```

File: scripts/get_summary.py [hunk 9]
```diff
@@ -1,33 +1,32 @@
...
-        result.average_get = statistics.mean(stats.get)
-        result.average_post = statistics.mean(stats.post)
-        result.average_total = statistics.mean(stats.get + stats.post)
-        result.error_percentage = (result.total_failures * 100) / result.total_requests
-
-        results_list.append([date, result])
-
-    return results_list
+        results = BenchmarkStats(folder)
+        yield Result(date, results)
```
Comment (MebinAbraham) [inline scripts/get_summary.py:29]: Not a blocker but `results` shadows the name from outer scope.

File: scripts/get_stress_test_summary.py [hunk 10]
```diff
@@ -1,37 +1,25 @@
...
         )
         sys.exit(1)
 
-    sorted_folders = sorted(glob(f"{folder}/*"))
-    result = get_results(sorted_folders)
+    sorted_folders = sorted(glob(f"{folders_path}/*"))
+    results = get_results(sorted_folders)
 
-    print(result)
+    for result in results:
```
Comment (berroar) [inline scripts/get_stress_test_summary.py:24]: Isn't the stress test script supposed to be outputting a summary at the top folder level? This is doing the same as get_summary now just without the date (unless I've misunderstood 😅)

Comment (MebinAbraham) [reply scripts/get_stress_test_summary.py:24]: Yup, you are right @berroar good spot.

**Pull request merged**